src/app/public/modules/lookup/lookup.component.spec.ts

```diff
@@ -364,16 +364,67 @@ describe('Lookup component', function () {
       }));
 
       it('should remove all but the first value when the mode is changed to single select', fakeAsync(() => {
-        component.friends = [{ name: 'Rachel' }, { name: 'Isaac'}];
+        component.friends = [{ name: 'Rachel' }, { name: 'Isaac' }];
         fixture.detectChanges();
 
         component.setSingleSelect();
         fixture.detectChanges();
         tick();
         fixture.detectChanges();
 
-        expect(lookupComponent.value).toEqual([{ name: 'Rachel'}]);
+        expect(lookupComponent.value).toEqual([{ name: 'Rachel' }]);
       }));
+
+      describe('form control interactions', function () {
+        it('should properly reset a form', fakeAsync(function () {
+          fixture.detectChanges();
+
+          performSearch('s', fixture);
+          selectSearchResult(0, fixture);
+
+          expect(lookupComponent.tokens.length).toBe(1);
+          expect(lookupComponent.value).toEqual([{ name: 'Isaac' }]);
+
+          component.resetForm();
+
+          expect(lookupComponent.tokens.length).toBe(0);
+          expect(lookupComponent.value).toEqual([]);
+        }));
+
+        it('should properly set a value through the form control', fakeAsync(function () {
+          fixture.detectChanges();
+
+          performSearch('s', fixture);
+          selectSearchResult(0, fixture);
+
+          expect(lookupComponent.tokens.length).toBe(1);
+          expect(lookupComponent.tokens[0].value).toEqual({ name: 'Isaac' });
+          expect(lookupComponent.value).toEqual([{ name: 'Isaac' }]);
+
+          component.setValue(0);
+
+          expect(lookupComponent.tokens.length).toBe(1);
+          expect(lookupComponent.tokens[0].value).toEqual({ name: 'Andy' });
+          expect(lookupComponent.value).toEqual([{ name: 'Andy' }]);
+        }));
+
+        it('should properly set a value through the form control', fakeAsync(function () {
```

Comment:
If this test is needed, it should have a different expectation than the one above it.

Reply:
The test is needed with the same expectations because it is testing two different ways of doing the same thing.

Angular allows you to set a form control via a `setValue` on the form control or `patchValue` on the form group. I agree that the tests are probably a little redundant but are there just to ensure everything is working properly. I can remove one of the tests from both describe blocks if you really want but I personally don't think it really hurts just to be a tad more careful.

Reply:
You know what - the more I think about it - you are probably right in this being overly careful. I'll remove the second test from both.